<?xml version="1.0"?>

<launch>
  <!--Provide option to set namespace from outside-->
  <arg name="namespace" default="/thor_mang"/>

    <!-- robot semantics for moveit -->
  <param name="/robot_description_semantic" textfile="$(find thor_mang_tud_moveit_config)/config/thor_mang_robot.srdf" />

  <group ns="$(arg namespace)">
    <!-- Load joint controller configurations from YAML file to parameter server -->
    <rosparam file="$(find thor_mang_ros_control)/config/thor_mang_states.yaml" command="load" />
    <rosparam file="$(find thor_mang_ros_control)/config/thor_mang_controllers.yaml" command="load" />
    <rosparam file="$(find thor_mang_ros_control)/config/thor_mang_position_controllers.yaml" command="load" />
    <rosparam file="$(find thor_mang_ros_control)/config/thor_mang_trajectory_controllers.yaml" command="load" />

    <!-- Load ft-sensor compensation data -->
    <rosparam file="$(find vigir_force_torque_compensation_lib)/config/thor_ft_compensation_data.yaml" command="load" />

    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"
      args="footstep_preview_controller
            joint_state_controller
            imu_sensor_controller
            force_torque_sensor_controller
            waist_lidar_controller"/>

    <!-- Load trajectory controllers -->
    <include file="$(find thor_mang_ros_control)/launch/load_trajectory_controllers.launch">
      <arg name="namespace" value="$(arg namespace)" />
    </include>

    <include  file = "$(find thor_mang_ros_control)/launch/hands/$(optenv THOR_RIGHT_HAND_TYPE r_stump)_controllers.launch" />
    <include  file = "$(find thor_mang_ros_control)/launch/hands/$(optenv THOR_LEFT_HAND_TYPE  l_stump)_controllers.launch"/>

    <node name="thor_mang_ros_controller" pkg="thor_mang_ros_control" type="thor_mang_ros_controller" respawn="true" output="screen">
      <rosparam file="$(find thor_mang_ros_control)/config/thor_mang_controller_config.yaml" command="load"/>
      <param name="thor_mang_ini_file" value="$(find thor_mang_ros_control)/config/thor_mang_config.ini" />
    </node>
  </group>
</launch>
